crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -713,6 +718,9 @@ impl<'db> TypeInferenceBuilder<'db> {
         let value_ty = self
             .types
             .expression_ty(assignment.value.scoped_ast_id(self.db, self.scope));
+        self.types
+            .expressions
+            .insert(target.scoped_ast_id(self.db, self.scope), value_ty);
```

Comment:
This fixes a bug where the type for the assignment's target expression was missing.